src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -607,12 +609,21 @@ def _update_zone(new: cos.CephObjectStore) -> cos.CephObjectStore:
             cos.CephObjectStore, 'cephobjectstores', name,
             _update_zone, _create_zone)
 
-    def apply_nfsgw(self, spec: NFSServiceSpec) -> str:
+    def apply_nfsgw(self, spec: NFSServiceSpec, mgr: 'RookOrchestrator') -> str:
         # TODO use spec.placement
         # TODO warn if spec.extended has entries we don't kow how
         #      to action.
         # TODO Number of pods should be based on the list of hosts in the
         #      PlacementSpec.
+        service_id = spec.service_id if spec.service_id is not None else ""
```

Comment:
`service_id` should never be empty string. Add a check for both None and empty string